src/SprykerMiddleware/Zed/Process/Business/Mapper/Mapper.php

```diff
@@ -54,19 +65,38 @@ public function map(array $payload): array
     protected function mapByRule(array $result, array $payload, string $key, $value): array
     {
         if (is_callable($value)) {
-            return $this->payloadManager->setValue($result, $key, $value($payload, $key));
+            return $this->mapCallable($result, $payload, $key, $value);
         }
         if (is_array($value)) {
             return $this->mapArray($result, $payload, $key, $value);
         }
         if (is_string($value) || is_int($value)) {
-            return $this->payloadManager
-                ->setValue($result, $key, $this->payloadManager->getValueByKey($payload, $value));
+            return $this->mapKey($result, $payload, $key, $value);
         }
 
         return $result;
     }
 
+    /**
+     * @param array $result
+     * @param array $payload
+     * @param string $key
+     * @param callable $value
+     *
+     * @return array
+     */
+    protected function mapCallable(array $result, array $payload, string $key, callable $value): array
+    {
+        $mappedValue = $value($payload, $key);
+        $this->logger->debug('Mapping', [
+            'operation' => 'Map callable',
+            'new_key' => $key,
+            'old_key' => $value,
+            'data' => $mappedValue,
+        ]);
+        return $this->payloadManager->setValue($result, $key, $mappedValue);
```

Comment:
Please don't use setter prefix (set-) to modify passing values. Setter should return void or $this.